### PR TITLE
Fixed Consul discovery with Hazelcast 4.0

### PR DESCRIPTION
**Pull request by amrsamii**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version=1.0-RC9
 description=hazelcast-consul-discovery-spi - Consul based discovery strategy SPI for Hazelcast enabled applications
 
-hazelcast_version=3.+
+hazelcast_version=4.0
```
Comment (bitsofinfo) [inline gradle.properties:4]: prob should be 4.+, no?

File: src/main/resources/doNothingRegistrator-example.xml [hunk 2]
```diff
@@ -1,31 +1,28 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<hazelcast id="hazelcast-consul-discovery" 
-           xsi:schemaLocation="http://www.hazelcast.com/schema/config hazelcast-config-3.6.xsd"
+<hazelcast id="hazelcast-consul-discovery"
+           xsi:schemaLocation="http://www.hazelcast.com/schema/config hazelcast-config-4.0.xsd"
            xmlns="http://www.hazelcast.com/schema/config"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
-    <group>
```
Comment (bitsofinfo) [inline src/main/resources/doNothingRegistrator-example.xml:6]: why is this removed, its used in the example/travis i thought no?

File: src/main/resources/explicitIpPortRegistrator-example.xml [hunk 3]
```diff
@@ -1,15 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <hazelcast id="hazelcast-consul-discovery" 
-           xsi:schemaLocation="http://www.hazelcast.com/schema/config hazelcast-config-3.6.xsd"
+           xsi:schemaLocation="http://www.hazelcast.com/schema/config hazelcast-config-4.0.xsd"
            xmlns="http://www.hazelcast.com/schema/config"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
            
     <!-- NOTE! any XML value below can be variables that will be fetched from
          -D system properties @see http://docs.hazelcast.org/docs/3.6-EA/manual/html-single/index.html#using-variables  -->
-    <group>
```
Comment (bitsofinfo) [inline src/main/resources/explicitIpPortRegistrator-example.xml:9]: ditto above, why is this removed, used in example/travis no?

**Review by bitsofinfo — COMMENTED** [review]
Please provide the necessary updates to the README on version support etc

File: gradle.properties (hunk 1 above)
Comment (amrsamii) [reply gradle.properties:4]: okay I will change it to 4.+

File: src/main/resources/doNothingRegistrator-example.xml (hunk 2 above)
Comment (amrsamii) [reply src/main/resources/doNothingRegistrator-example.xml]: The <group> tag has been removed in 4.0 and replaced by <cluster-name>

File: src/main/resources/explicitIpPortRegistrator-example.xml (hunk 3 above)
Comment (amrsamii) [reply src/main/resources/explicitIpPortRegistrator-example.xml]: The <group> tag has been removed in 4.0 and replaced by <cluster-name>

File: src/main/resources/doNothingRegistrator-example.xml (hunk 2 above)
Comment (amrsamii) [reply src/main/resources/doNothingRegistrator-example.xml:6]: The group tag has been removed in 4.0 and replaced by cluster-name

File: src/main/resources/explicitIpPortRegistrator-example.xml (hunk 3 above)
Comment (amrsamii) [reply src/main/resources/explicitIpPortRegistrator-example.xml:9]: The group tag has been removed in 4.0 and replaced by cluster-name

**Comment by amrsamii** [comment]
@bitsofinfo could you please specify which section of README to be updated?

**Comment by amrsamii** [comment]
@bitsofinfo Thank you for merging, but what about README?

**Comment by bitsofinfo** [comment]
@amrsamii https://github.com/bitsofinfo/hazelcast-consul-discovery-spi/releases/tag/2.0-RC1

**Comment by amrsamii** [comment]
okay it is done now. I am really happy that I contributed to your repo.
